src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -154,22 +133,36 @@ export default class GooglePayPaymentProcessor {
                     throw new MissingDataError(MissingDataErrorType.MissingCheckout);
                 }
 
-                const testMode = paymentMethod.config.testMode;
+                const { testMode } = paymentMethod.config;
 
                 return Promise.all([
                     this._googlePayScriptLoader.load(),
                     this._googlePayInitializer.initialize(checkout, paymentMethod, hasShippingAddress),
-                ])
-                    .then(([googlePay, googlePayPaymentDataRequest]) => {
-                        this._googlePaymentsClient = this._getGooglePayClient(googlePay, testMode);
-                        this._googlePaymentDataRequest = googlePayPaymentDataRequest;
-                    })
-                    .catch((error: Error) => {
-                        throw error;
-                    });
+                ]).then(([googlePay, paymentDataRequest]) => {
+                        this._googlePayClient = this._getGooglePayClient(googlePay, testMode);
+                        this._paymentDataRequest = paymentDataRequest;
```

Comment:
So you're caching the `paymentDataRequest` value here and using it in `displayWallet `. I don't think you should be doing that if you can avoid it because you're passing in the `checkout` object in order to retrieve the request data. But the shopper could be changing the checkout object before `displayWallet ` is called (i.e.: applying a coupon code or gift certificate, which would change the grand total). Therefore, you should defer the creation of `paymentDataRequest` until you need it, which is when `displayWallet` is called.

Reply:
Hey @davidchin ! So far, we've tried everything to address this as you asked but is seems something is not working properly in the dispatcher that causes to get with a weird state on checkout  object.

In terms of your concern we're attaching a [vIdeo](https://drive.google.com/open?id=1prcVysEbDp1zy1p543az2m0ouiL2B2m1)  that proofs the transaction gets authorized with the right amount (event if we apply a coupon or a gift certificate)

Reply:
Look, if you really can't get it work, I'm happy for you to keep it like this for now, since you've tested everything.